common/djangoapps/edxmako/shortcuts.py

```diff
@@ -70,9 +71,13 @@ def marketing_link(name):
     elif not enable_mktg_site and name in link_map:
         # don't try to reverse disabled marketing links
         if link_map[name] is not None:
-            return reverse(link_map[name])
+            host_name = get_current_request_hostname()
+            if all([host_name, 'edge' in host_name, 'http' in link_map[name]]):
```

Comment:
Test will fail again with this, you should do it like this `host_name and 'edge' in host_name`.